oasis-reference/src/main/kotlin/com/empowerops/volition/ref_oasis/OptimizerController.kt

```diff
@@ -295,4 +310,59 @@ class OptimizerController {
         }
     }
 
-}
\ No newline at end of file
+    @Subscribe
+    fun onPaused(event: PausedEvent) = GlobalScope.launch(Dispatchers.JavaFx) {
+        uiStateMachine.transferTo(State.Paused)
```

Comment:
This state thing is a bit redundant, my original thought is make the state machine do the UI update but I haven't figure out a good way to do that. It will be like rebindViewToState() and in pause state do the action follow, however, we also have the case where different behavior we in the case where we transfer from a state. I think the solution to that we re-assign every button state. Then we have much less to worry about.

Reply:
It will be like
```
@Subscribe
fun onStateChagned(event: StateChangedEvent){
    stateMachine.transerTo(event)
    rebind()
}

fun rebind(){
    when(state){
        ...
        paused -> {btn1.setText(...); btn1.setGraphic(...); btn1.setDisabled(); label1.setText(...)}
        pausing -> ...
        ...
    }
}
```

Reply:
Somehow my original comment got deleted.

Regarding your original question:

> It will be like rebindViewToState() and in pause state do the action follow, however, we also have the case where different behavior we in the case where we transfer from a state. I think the solution to that we re-assign every button state. Then we have much less to worry about.

Yeah, thats how I've generally written controllers and its one of the ways you can get the granularity of events down so that you only need course-grained events (IE, things like `ModelChangedEvent` rather than `ModelAddedNonZeroTimeoutEvent` --the former is very general and just triggers basic `rebindViewToModel()` flows, the latter is very specific and will often only trigger very specific updates, which can be bug-prone)